src/TypesFinder/FindTypeFromAst.php

```diff
@@ -19,10 +19,16 @@ class FindTypeFromAst
      */
     public function __invoke($astType, LocatedSource $locatedSource, $namespace = '')
     {
-        $context = (new ContextFactory())->createForNamespace(
-            $namespace,
-            $locatedSource->getSource()
-        );
+        static $contexts = [];
```

Comment:
This class is instantiated everytime it is required, and the AST is parsed each time, even when the same namespace is used.